rotkehlchen/db/dbhandler.py

```diff
@@ -454,10 +473,11 @@ def upgrade_db_sqlcipher_3_to_4(self, password: str) -> Tuple[bool, str]:
 
         return success, msg
 
-    def disconnect(self) -> None:
-        if hasattr(self, 'conn') and self.conn:
-            self.conn.close()
-            self.conn = None
+    def disconnect(self, conn_attribute: Literal['conn', 'conn_transient'] = 'conn') -> None:
+        if conn_attribute and hasattr(self, conn_attribute) and getattr(self, conn_attribute):
+            conn = getattr(self, conn_attribute)
```

Comment:
You are doing too many unneeded stuff here. conn_attribute, has attr, getattr and then again getattr?!

```suggestion
        conn = getattr(self, conn_attribute, None)
        if conn:
```

`conn_attribute` CAN NOT be anything other than `conn` and `conn_transient`. Your typing also shows this.